src/sentry/grouping/strategies/newstyle.py

```diff
@@ -510,7 +543,9 @@ def single_exception(exception, context, **meta):
     rv = {}
 
     for variant, stacktrace_component in stacktrace_variants.items():
-        values = [stacktrace_component, type_component]
+        values = [stacktrace_component]
+        if not context["hierarchical_grouping"] or variant != "app-depth-1":
```

Comment:
Trying to understand: Why does variant "app-depth-1" get special treatment here?

Reply:
we should probably remove this. it was to work around in a bug that affected a specific customer but eventually we will fix it in the native sdk